src/Witnesses.h

```diff
@@ -72,6 +75,7 @@ class InvalidityWitness {
     void setDerivation(Derivation derivation_) { derivation = std::move(derivation_); }
 
     [[nodiscard]] Derivation const & getDerivation() const { return derivation; }
+    [[nodiscard]] Derivation & getDerivation() { return derivation; }
```

Comment:
Hmm, why 2 functions with different modifiers?

Reply:
Because I needed also the non-`const` version. That is a pretty common pattern.

Reply:
Ok, gotcha